js/app/interfaces/card.js

```diff
@@ -365,18 +365,25 @@ const Card = new Lang.Interface({
      *
      * Sets up a frame to show the model's thumbnail uri.
      */
-    set_thumbnail_frame_from_model: function (frame) {
+    set_thumbnail_frame_from_model: function (frame, background_size='cover') {
         frame.visible = false;
         if (!this.model.thumbnail_uri)
             return;
+        if (background_size === 'center') {
+            let im = new Gtk.Image({
```

Comment:
Maybe we just pull the trigger on https://phabricator.endlessm.com/T12679 whether or not gsk finds it way into our gtk this milestone. Then you could try to do all this with css and keep things more responsive.